File: backend/routers/college.py
"""
College router - College tasks endpoints
"""
from fastapi import APIRouter, Depends, Query
from typing import Optional
from uuid import UUID
import math

from schemas.college import (
    CreateCollegeTaskRequest,
    UpdateCollegeTaskRequest,
    CollegeTaskResponse
)
from services.college_service import college_service
from core.auth import get_current_user, CurrentUser
from core.responses import Response, PaginatedResponse

router = APIRouter()


@router.post("/tasks", response_model=Response[CollegeTaskResponse])
async def create_task(
    request: CreateCollegeTaskRequest,
    current_user: CurrentUser = Depends(get_current_user)
):
    """Create a new college task"""
    task = await college_service.create_task(current_user.user_id, request)
    return Response(
        success=True,
        message="College task created successfully",
        data=task
    )


@router.get("/tasks", response_model=PaginatedResponse[CollegeTaskResponse])
async def get_tasks(
    status: Optional[str] = Query(None, description="Filter by status: ongoing, completed, pending, overdue"),
    page: int = Query(1, ge=1, description="Page number"),
    limit: int = Query(20, ge=1, le=100, description="Items per page"),
    current_user: CurrentUser = Depends(get_current_user)
):
    """Get college tasks with pagination"""
    tasks, total = await college_service.get_tasks(
        user_id=current_user.user_id,
        status=status,
        page=page,
        limit=limit
    )
    
    return PaginatedResponse(
        success=True,
        message="College tasks retrieved successfully",
        data=tasks,
        page=page,
        limit=limit,
        total=total,
        total_pages=math.ceil(total / limit)
    )


@router.get("/tasks/{task_id}", response_model=Response[CollegeTaskResponse])
async def get_task(
    task_id: UUID,
    current_user: CurrentUser = Depends(get_current_user)
):
    """Get college task by ID"""
    task = await college_service.get_task_by_id(current_user.user_id, task_id)
    return Response(
        success=True,
        message="College task retrieved successfully",
        data=task
    )


@router.put("/tasks/{task_id}", response_model=Response[CollegeTaskResponse])
async def update_task(
    task_id: UUID,
    request: UpdateCollegeTaskRequest,
    current_user: CurrentUser = Depends(get_current_user)
):
    """Update college task"""
    task = await college_service.update_task(current_user.user_id, task_id, request)
    return Response(
        success=True,
        message="College task updated successfully",
        data=task
    )


@router.delete("/tasks/{task_id}", response_model=Response[dict])
async def delete_task(
    task_id: UUID,
    current_user: CurrentUser = Depends(get_current_user)
):
    """Delete college task"""
    await college_service.delete_task(current_user.user_id, task_id)
    return Response(
        success=True,
        message="College task deleted successfully",
        data={}
    )

File: src/api/college.ts
/**
 * College API Service
 */
import { apiClient, PaginatedResponse } from './client';

export interface CollegeTask {
  id: string;
  user_id: string;
  title: string;
  description?: string;
  type: string;
  status: string;
  due_date?: string;
  priority?: string;
  subject?: string;
  tags: string[];
  created_at: string;
  updated_at: string;
}

export interface CreateCollegeTaskRequest {
  title: string;
  description?: string;
  type: string;
  status?: string;
  due_date?: string;
  priority?: string;
  subject?: string;
  tags?: string[];
}

export interface UpdateCollegeTaskRequest {
  title?: string;
  description?: string;
  type?: string;
  status?: string;
  due_date?: string;
  priority?: string;
  subject?: string;
  tags?: string[];
}

export const collegeApi = {
  /**
   * Get college tasks with pagination
   */
  getTasks: async (params?: {
    status?: string;
    page?: number;
    limit?: number;
  }): Promise<PaginatedResponse<CollegeTask>> => {
    const response = await apiClient.get<PaginatedResponse<CollegeTask>>('/college/tasks', {
      params,
    });
    return response.data;
  },

  /**
   * Get task by ID
   */
  getTask: async (id: string): Promise<CollegeTask> => {
    const response = await apiClient.get<CollegeTask>(`/college/tasks/${id}`);
    return response.data;
  },

  /**
   * Create college task
   */
  createTask: async (data: CreateCollegeTaskRequest): Promise<CollegeTask> => {
    const response = await apiClient.post<CollegeTask>('/college/tasks', data);
    return response.data;
  },

  /**
   * Update college task
   */
  updateTask: async (id: string, data: UpdateCollegeTaskRequest): Promise<CollegeTask> => {
    const response = await apiClient.put<CollegeTask>(`/college/tasks/${id}`, data);
    return response.data;
  },

  /**
   * Delete college task
   */
  deleteTask: async (id: string): Promise<void> => {
    await apiClient.delete(`/college/tasks/${id}`);
  },
};
